From 526d447ca608995abb0c58b94c6a912fe23ea089 Mon Sep 17 00:00:00 2001
From: Erik Bengtsson <erik.bengtsson@flir.com>
Date: Tue, 15 Mar 2022 11:00:22 +0100
Subject: [PATCH] ov5640: set fixed values to reg when reading type

This would cause OTP read not to work. Set fixed values
to know it will work as expected.
---
 .../media/platform/mxc/capture/ov5640_v2.c    | 20 ++-----------------
 1 file changed, 2 insertions(+), 18 deletions(-)

diff --git a/drivers/media/platform/mxc/capture/ov5640_v2.c b/drivers/media/platform/mxc/capture/ov5640_v2.c
index 43144e2565c3..8e90297cd25b 100644
--- a/drivers/media/platform/mxc/capture/ov5640_v2.c
+++ b/drivers/media/platform/mxc/capture/ov5640_v2.c
@@ -1381,22 +1381,6 @@ static s32 ov5640_read_reg(u16 reg, u8 *val)
 	return u8RdVal;
 }
 
-static s32 ov5640_mod_reg(u16 reg, u8 mask, u8 val)
-{
-	u8 readval;
-	s32 ret;
-
-	ret = ov5640_read_reg(reg, &readval);
-	if (ret)
-		return ret;
-
-	readval &= ~mask;
-	val &= mask;
-	val |= readval;
-
-	return ov5640_write_reg(reg, val);
-}
-
 #ifdef CONFIG_VIDEO_ADV_DEBUG
 static int ov5640_get_register(struct v4l2_subdev *sd,
 					struct v4l2_dbg_register *reg)
@@ -1800,13 +1784,13 @@ static int ov5640_get_sensor_model(u8 *sensor_model, int n)
 		return -1;
 	}
 
-	retval = ov5640_mod_reg(OV5640_SYSTEM_RESET00, BIT(4), 0);
+	retval = ov5640_write_reg(OV5640_SYSTEM_RESET00, 0x20);
 	if (retval < 0) {
 		pr_err("ov5640: failed to enable OTP module\n");
 		return -1;
 	}
 
-	retval = ov5640_mod_reg(OV5640_CLOCK_ENABLE00, BIT(4), BIT(4));
+	retval = ov5640_write_reg(OV5640_CLOCK_ENABLE00, 0xDF);
 	if (retval < 0) {
 		pr_err("ov5640: failed to enable OTP clock\n");
 		return -1;
-- 
2.17.1

